main.go

```diff
@@ -246,16 +246,29 @@ func main() {
 	listenAddr := fmt.Sprintf("%s:%d", cfg.Listen.Address, cfg.Listen.Port)
 	endpoint := cfg.Listen.MetricsEndpointOrDefault()
 
-	fmt.Printf("running HTTP server on address %s, serving metrics at %s\n", listenAddr, endpoint)
 
 	nsHandler := promhttp.InstrumentMetricHandler(
 		prometheus.DefaultRegisterer, promhttp.HandlerFor(nsGatherers, promhttp.HandlerOpts{}),
 	)
 
 	http.Handle(endpoint, nsHandler)
 
-	if err := http.ListenAndServe(listenAddr, nil); err != nil {
-		fmt.Printf("error while starting HTTP server: %s", err.Error())
+	if cfg.Listen.TLS.Enable {
+		if _, err := os.Stat(cfg.Listen.TLS.CertFile); err != nil {
+			fmt.Printf("Error loading certificate: %s", err.Error())
+		}
+		if _, err := os.Stat(cfg.Listen.TLS.KeyFile); err != nil {
+			fmt.Printf("Error loading key: %s", err.Error())
+		}
```

Comment:
If one of these errors occurs, the process should exit, and _not_ continue trying to start a server.